PSICompile/ILCodeGen.cs

```diff
@@ -67,7 +67,17 @@ public class ILCodeGen : Visitor {
       if (w.NewLine) Out ("    call void [System.Console]System.Console::WriteLine ()");
    }
    
-   public override void Visit (NIfStmt f) => throw new NotImplementedException ();
+   public override void Visit (NIfStmt f) {
+      string lab1 = NextLabel (), lab2 = NextLabel (), lab3 = NextLabel ();
+      f.Condition.Accept (this);
+      Out ($"    brfalse {lab2}");
+      Out ($"  {lab1}:");
+      f.IfPart.Accept (this);
+      Out ($"    br {lab3}");
+      Out ($"  {lab2}:");
+      f.ElsePart?.Accept (this);
+      Out ($"  {lab3}:");
+   }

```

Comment:
This works, but uses too many labels. Let's take an if statement with no else part (for example like this):
`if (2 < 3) then WriteLn (2);`
You are generating this code:
```
    ldc.i4 2
    ldc.i4 3
    clt
    brfalse IL_0002
  IL_0001:
    ldc.i4 2
    call void [System.Console]System.Console::Write (int32)
    call void [System.Console]System.Console::WriteLine ()
    br IL_0003
  IL_0002:
  IL_0003:
```
I am expecting this code:
```
    ldc.i4 2
    ldc.i4 3
    clt
    brfalse IL_0000
    ldc.i4 2
    call void [System.Console]System.Console::Write (int32)
    call void [System.Console]System.Console::WriteLine ()
  IL_0001:
```

Reply:
Generated the code as above:
![image](https://github.com/tarydon/PSI/assets/125174306/4e3c8749-9b61-43bd-b048-0f80b9900656)

but there is a compiler error of missing IL_0000 in the forward reference.
![image](https://github.com/tarydon/PSI/assets/125174306/e3c0d774-ad4e-4862-97b1-56acffa5a811)

correct me, if i am doing anything wrong!

Reply:
hope it's typo error at brfalse:
![image](https://github.com/tarydon/PSI/assets/125174306/75c3c664-7af3-4cbf-b44f-5a260c2b18f3)

now updated the code as follows:
![image](https://github.com/tarydon/PSI/assets/125174306/22437dcb-c6de-44d8-9ee3-5732d8fee92f)